src/renderers/shared/fiber/getCurrentOwnerName.js

```diff
@@ -12,13 +12,38 @@
 
 'use strict';
 
+import type { Fiber } from 'ReactFiber';
+
 var ReactCurrentOwner = require('ReactCurrentOwner');
+var ReactTypeOfWork = require('ReactTypeOfWork');
 var getComponentName = require('getComponentName');
 
+var {
+  IndeterminateComponent,
+  FunctionalComponent,
+  ClassComponent,
+} = ReactTypeOfWork;
+
 module.exports = function() : null | string {
-  if (ReactCurrentOwner.current) {
-    return getComponentName(ReactCurrentOwner.current);
-  } else {
+  let owner = ReactCurrentOwner.current;
+  if (owner == null) {
     return null;
   }
+  // In Stack, current owner is always a composite.
+  if (typeof owner.tag !== 'number') {
+    return getComponentName(owner);
+  }
+  // In Fiber, current owner could be any fiber.
+  let fiber = ((owner : any) : Fiber);
```

Comment:
Owner could also be a Stack instance. https://github.com/facebook/react/pull/8458